src/types.jl

```diff
@@ -160,6 +160,23 @@ struct RGB4{T<:Fractional} <: AbstractRGB{T}
 end
 RGB4(r::T, g::T, b::T) where {T<:Fractional} = RGB4{T}(r, g, b)
 
+"""`RGB16` is Red-Green-Blue colorspace represented by two bytes (16 bits).
+
+1        |1          |        0
+5 4 3 2 1|0 9 8 7 6 5|4 3 2 1 0 
+---------|-----------|---------
+R R R R R|G G G G G G|B B B B B
+"""
+struct RGB16
+    value::UInt16
+end
+
+function RGB16(r, g, b)
+    RGB16(r << 11 + g << 5 + b)
+end
+
+UInt16(c::RGB16) = c.value
```

Comment:
`UInt16(x)` converts a number while preserving its value, but here you're thinking of this as a bit pattern. Consequently this should be `reinterpret(::Type{UInt16}, c::RGB16) = c.value`.